src/features/utils/instrument-base.js

```diff
@@ -0,0 +1,37 @@
+import { registerDrain } from "../../common/drain/drain"
+import { FeatureBase } from "./feature-base"
+import { onWindowLoad } from '../../common/window/load'
+import { isWorkerScope } from "../../common/util/global-scope"
+import { FEATURE_NAMES } from "../../loaders/features/features"
+
+export class InstrumentBase extends FeatureBase {
+  constructor(agentIdentifier, aggregator, featureName, auto = true) {
+    super(agentIdentifier, aggregator, featureName)
+    this.completed = new Promise((resolve, reject) => {
+      this.resolve = resolve
+      this.reject = reject
+    })
+    this.hasAggregator = false
+    this.auto = auto
+
+    if (auto && featureName !== FEATURE_NAMES.sessionTrace) registerDrain(agentIdentifier, featureName)
+  }
+
+  importAggregator() {
+    try {
+      if (this.hasAggregator || !this.auto) return
```

Comment:
Is `auto` for features being set somewhere else? Reading this and in relation to the previous above comment, I would've expected page actions to not import due to this short circuit. But your tests are passing, so I'm confused.

Reply:
auto is a constructor arg, basically only used by tests at this point
```
  constructor(agentIdentifier, aggregator, featureName, auto = true) { ... }
  ```